modules/gapi/src/executor/gstreamingexecutor.cpp

```diff
@@ -1126,14 +1126,16 @@ cv::gimpl::GStreamingExecutor::GStreamingExecutor(std::unique_ptr<ade::Graph> &&
     m_sink_queues   .resize(proto.out_nhs.size(), nullptr);
     m_sink_sync     .resize(proto.out_nhs.size(), -1);
 
-    // Very rough estimation to limit internal queue sizes.
+    // Very rough estimation to limit internal queue sizes if not specified by the user.
     // Pipeline depth is equal to number of its (pipeline) steps.
-    const auto queue_capacity = 3*std::count_if
-        (m_gim.nodes().begin(),
-         m_gim.nodes().end(),
-         [&](ade::NodeHandle nh) {
-            return m_gim.metadata(nh).get<NodeKind>().k == NodeKind::ISLAND;
-         });
+    auto has_queue_capacity = cv::gapi::getCompileArg<cv::gapi::queue_capacity>(m_comp_args);
+    const auto queue_capacity = has_queue_capacity ? has_queue_capacity->capacity :
```

Comment:
What if uses passes 0 there? He can, for sure!

Reply:
BTW, does it make sense to use `int` instead of `size_t` to allow user pass `-1` as well ?

Reply:
Not sure that extra logic like checking capacity equal to zero is required, but  I can add

Reply:
Needed ?

Reply:
No need for -1. There're better ways to indicate some extra meaning of the variable besides its numeric value.

Reply:
This is still open.

Reply:
Done